keanu-project/src/test/java/io/improbable/keanu/e2e/regression/RadonHeirarchicalRegression.java

```diff
@@ -0,0 +1,179 @@
+package io.improbable.keanu.e2e.regression;
+
+import io.improbable.keanu.DeterministicRule;
+import io.improbable.keanu.algorithms.variational.optimizer.gradient.GradientOptimizer;
+import io.improbable.keanu.model.regression.RegressionModel;
+import io.improbable.keanu.model.regression.RegressionRegularization;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.util.csv.ReadCsv;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.HalfGaussianVertex;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class RadonHeirarchicalRegression {
+
+    @Rule
+    public DeterministicRule rule = new DeterministicRule();
+
+    private List<Data> radonData;
+
+    @Before
+    public void readRadonCSV() {
+        radonData = ReadCsv.fromResources("data/datasets/radon/radon.csv")
+            .asRowsDefinedBy(Data.class)
+            .load(true);
+    }
+
+    @Test
+    public void canPerformSimpleLinearRegression() {
+        RegressionModel model = linearRegression(radonData);
+        Assert.assertTrue(model.getWeight(0) > -0.7 && model.getWeight(0) < -0.4);
+        Assert.assertTrue(model.getIntercept() > 1.2 && model.getIntercept() < 1.5);
+    }
+
+    @Test
+    public void canPerformRegressionWithOneHeirarchy() {
+        buildHeirarchicalNetwork(radonData, 1);
+    }
+
+    @Test
+    public void canPerformRegressionWithTwoHeirarchies() {
```

Comment:
Do we have any idea why the optimiser is struggling so much with the number of latents?  I wonder whether it's a case of it falling off in to the Wilderness of the HalfGaussian?  We've seen our gradient optimiser really struggle with those limited support distributions in the past?

Reply:
Nah it struggles with a uniform too. It's just not a very good optimiser and requires a lot of `assistance` to find the right answer as the dimensionality goes up